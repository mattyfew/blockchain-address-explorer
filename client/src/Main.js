import React from 'react';
import { ScaleLoader } from 'react-spinners';


export default function Main(props) {
    return (
        <div id="main">
            <header>
                <h2>Enter a BTC address to view transaction history</h2>
                <p className="address">Try <strong>1F1tAaz5x1HUXrCNLbtMDqcw6o5GNn4xqX</strong></p>
            </header>
            <div id="search-bar">
                {props.notValid && <div className="error">There is something wrong with that address</div>}
                {props.loading &&
                    <ScaleLoader
                      margin={'20px 5px 5px 0'}
                      color={'#123abc'}
                      loading={props.loading}
                    />
                }
                <input
                    onChange={props.onAddressChange}
                    id="search-bar-input"
                    placeholder="enter BTC address"
                    autoFocus/>
                <button onClick={props.submitAddress}>Search</button>
            </div>
        </div>
    )
}
