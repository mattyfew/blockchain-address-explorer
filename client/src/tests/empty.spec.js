import React from 'react';
import Main from '../Main'
import renderer from 'react-test-renderer';

import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({adapter: new Adapter()});

describe('App', () => {

    it('renders without crashing', () => {
        enzyme.shallow(<Main/>);
    });

    const component = renderer.create(
        <Main
            onAddressChange={() => {}}
            submitAddress={() => {}}/>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});










// describe('Address type', () => {
//     it('should be a main tag', () => {
//         console.log("RAAAWWWRRRR")
//
//         const renderer = new ShallowRenderer();
//         renderer.render(<Address addressInfo={sample}/>)
//
//         const actual = renderer.getRenderOutput().type;
//         const expected = 'main'
//         expect(actual).toEqual(expected)
//
//     })
// })
//
//
//  describe('createKey', () => {
//      it('should create a unique key for each rendered transaction', () => {
//
//          const renderer = TestUtils.createRenderer()
//          renderer.render(<Main />)
//          const output = renderer.getRenderOutput()
//          console.log(output)
//
//           expect(actual).toEqual(expected)
//      })
//  })
