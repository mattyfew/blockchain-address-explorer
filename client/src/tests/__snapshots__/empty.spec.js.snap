// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  id="main"
>
  <header>
    <h2>
      Enter a BTC address to view transaction history
    </h2>
    <p
      className="address"
    >
      Try 
      <strong>
        1F1tAaz5x1HUXrCNLbtMDqcw6o5GNn4xqX
      </strong>
    </p>
  </header>
  <div
    id="search-bar"
  >
    <input
      autoFocus={true}
      id="search-bar-input"
      onChange={[Function]}
      placeholder="enter BTC address"
    />
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
